### PR TITLE
Add DLC to Launch Handler migration info

### PR DESCRIPTION
**Pull request by joycetoh8**
Changes proposed in this pull request:

- Add information for migrating from Declarative Link Handling to Launch Handler API
- Introduces new Link Capturing user-enabled feature in Chrome

### CODE REVIEW

File: src/site/content/en/blog/declarative-link-capturing/index.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ subhead: |
 authors:
   - thomassteiner
 date: 2021-05-19
-updated: 2021-12-14
+updated: 2021-12-20
```
Comment (tomayac) [inline src/site/content/en/blog/declarative-link-capturing/index.md:12]:
```suggestion
updated: 2021-12-21
```

File: src/site/content/en/blog/declarative-link-capturing/index.md [hunk 2]
```diff
@@ -152,6 +154,51 @@ possible for the site to automatically trigger it. Second, the capability to foc
 window on its own domain and fire an event containing the clicked URL. This is intended to allow the
 site to navigate an existing window to a new page, overriding the default HTML navigation flow.
 
+## Migrate to Launch Handler API {: #migration }
+
+The Declarative Link Capturing API [origin trial](https://developer.chrome.com/origintrials/#/view_trial/4285175045443026945)
+is set to [expire on March 30, 2022](https://groups.google.com/a/chromium.org/g/blink-dev/c/2c4bul4V3GQ/m/Anluh1txBQAJ)
+for Chrome M97 and below. It will be replaced by a set of [new features and APIs](https://docs.google.com/document/d/1w9qHqVJmZfO07kbiRMd9lDQMW15DeK5o-p-rZyL7twk/edit)
```
Comment (tomayac) [inline src/site/content/en/blog/declarative-link-capturing/index.md:161]:
```suggestion
for Chromium&nbsp;97 and below. It will be replaced by a set of [new features and APIs](https://docs.google.com/document/d/1w9qHqVJmZfO07kbiRMd9lDQMW15DeK5o-p-rZyL7twk/edit)
```

File: src/site/content/en/blog/declarative-link-capturing/index.md [hunk 3]
```diff
@@ -152,6 +154,51 @@ possible for the site to automatically trigger it. Second, the capability to foc
...
+in Chrome 98 and above, which includes user-enabled link capturing and [Launch Handler API](https://github.com/WICG/sw-launch/blob/main/launch_handler.md).
```
Comment (tomayac) [inline src/site/content/en/blog/declarative-link-capturing/index.md:162]:
```suggestion
in Chromium&nbsp;98 and above, which includes user-enabled link capturing and [Launch Handler API](https://github.com/WICG/sw-launch/blob/main/launch_handler.md).
```

File: src/site/content/en/blog/declarative-link-capturing/index.md [hunk 4]
```diff
@@ -152,6 +154,51 @@ possible for the site to automatically trigger it. Second, the capability to foc
...
+
+### Link Capturing
+
+In Chrome 98, automatic link capturing is now a user opt-in behaviour rather than granted at install
```
Comment (tomayac) [inline src/site/content/en/blog/declarative-link-capturing/index.md:166]:
```suggestion
In Chromium&nbsp;98, automatic link capturing is now a user opt-in behavior rather than granted at install
```

File: src/site/content/en/blog/declarative-link-capturing/index.md [hunk 5]
```diff
@@ -152,6 +154,51 @@ possible for the site to automatically trigger it. Second, the capability to foc
...
+time to a web app. To enable link capturing, a user needs to launch an installed app from the browser
+using “Open with” and choose “Remember my choice”:
```
Comment (tomayac) [inline src/site/content/en/blog/declarative-link-capturing/index.md:168]:
```suggestion
using **Open with** and choose **Remember my choice**:
```

File: src/site/content/en/blog/declarative-link-capturing/index.md [hunk 6]
```diff
@@ -152,6 +154,51 @@ possible for the site to automatically trigger it. Second, the capability to foc
...
+
+{% Img src="image/ttTommHYbJXsEL29zNB1wXBvH4z1/rbFk5LtzHMlN3zRVpekf.png", alt="Example of an installed app's 'Open with' setting with the 'Remember my choice' option enabled.", width="385", height="277" %}
+
+Alternatively, users can switch link capturing on or off for a specific web app in the app management settings page:
```
Comment (tomayac) [inline src/site/content/en/blog/declarative-link-capturing/index.md:172]:
```suggestion
Alternatively, users can switch link capturing on or off for a specific web app in the app management settings page.
```

File: src/site/content/en/blog/declarative-link-capturing/index.md [hunk 7]
```diff
@@ -152,6 +154,51 @@ possible for the site to automatically trigger it. Second, the capability to foc
...
+
+{% Img src="image/ttTommHYbJXsEL29zNB1wXBvH4z1/WJ2KPqFUjqJABNYQUEN9.png", alt="Example of an installed app's settings page.", width="800", height="449" %}
+
+Link capturing is a Chrome OS only feature for now; support for Windows/Mac/Linux is still in progress.
```
Comment (tomayac) [inline src/site/content/en/blog/declarative-link-capturing/index.md:176]:
```suggestion
Link capturing is a Chrome&nbsp;OS-only feature for now; support for Windows, macOS, and Linux is still in progress.
```

File: src/site/content/en/blog/declarative-link-capturing/index.md [hunk 8]
```diff
@@ -152,6 +154,51 @@ possible for the site to automatically trigger it. Second, the capability to foc
...
+
+### Launch Handler API
+
+The control of an incoming navigation is migrated to Launch Handler API, which allows web apps to decide
+how a web app launches at various situations, such as link capturing, share target or file handling, etc.
+To migrate from the Declarative Link Capturing API to the Launch Handler API:
+
+1. Register your site for the [Launch Handler origin trial](https://developer.chrome.com/origintrials/#/view_trial/2978005253598740481) and place the origin trial key into your web app.
+1. Add a `"launch_handler"` entry to your site’s manifest.
```
Comment (tomayac) [inline src/site/content/en/blog/declarative-link-capturing/index.md:185]:
```suggestion
1. Add a `"launch_handler"` entry to your site's manifest.
```

File: src/site/content/en/blog/declarative-link-capturing/index.md [hunk 9]
```diff
@@ -152,6 +154,51 @@ possible for the site to automatically trigger it. Second, the capability to foc
...
+   -  If you are using `"capture_links": "new-client"` then add:`"launch_handler": { "route_to": "new-client" }`
+   -  If you are using `"capture_links": "existing-client-navigate"` then add: `"launch_handler": { "route_to": "existing-client" }`
+   -  If you wanted to use `"capture_links": "existing-client-event"` (which was never implemented in the Declarative Link Capturing origin trial) then add:
```
Comment (tomayac) [inline src/site/content/en/blog/declarative-link-capturing/index.md:188]:
```suggestion
   -  If you are using `"capture_links": "new-client"`, then add:`"launch_handler": { "route_to": "new-client" }`
   -  If you are using `"capture_links": "existing-client-navigate"`, then add: `"launch_handler": { "route_to": "existing-client" }`
   -  If you wanted to use `"capture_links": "existing-client-event"` (which was never implemented in the Declarative Link Capturing origin trial), then add:
```

File: src/site/content/en/blog/declarative-link-capturing/index.md [hunk 10]
```diff
@@ -152,6 +154,51 @@ possible for the site to automatically trigger it. Second, the capability to foc
...
+      ```json
+      "launch_handler": {
+        "route_to": "existing-client",
+        "navigate_existing_client": "never",
+      }
```
Comment (tomayac) [inline src/site/content/en/blog/declarative-link-capturing/index.md:193]:
```suggestion
{
   "launch_handler": {
     "route_to": "existing-client",
     "navigate_existing_client": "never"
   }
} 
```

File: src/site/content/en/blog/declarative-link-capturing/index.md [hunk 11]
```diff
@@ -152,6 +154,51 @@ possible for the site to automatically trigger it. Second, the capability to foc
...
+      ```json
+      "launch_handler": {
+        "route_to": "existing-client",
+        "navigate_existing_client": "never",
+      } 
+      ```
+      
+      **Note**: With this option, pages in your app scope will no longer navigate automatically when a link navigation is captured. You must handle the `LaunchParams` in JavaScript via the `window.launchQueue.setConsumer()` API to enable navigation.
+
+1. Keep the `capture_links` field and Declarative Link Capturing origin trial registration until March 30, 2022. This will ensure users on Chrome 97 and below can still launch the web app at a captured link.
```
Comment (tomayac) [inline src/site/content/en/blog/declarative-link-capturing/index.md:198]:
```suggestion
1. Keep the `capture_links` field and Declarative Link Capturing origin trial registration until March 30, 2022. This will ensure users on Chromium&nbsp;97 and below can still launch the web app at a captured link.
```

Comment (tomayac) [inline src/site/content/en/blog/declarative-link-capturing/index.md:198]:
```suggestion
1. Keep the `capture_links` field and Declarative Link Capturing origin trial registration until March&nbsp;30, 2022. This will ensure users on Chrome 97 and below can still launch the web app at a captured link.
```

File: src/site/content/en/blog/declarative-link-capturing/index.md [hunk 12]
```diff
@@ -152,6 +154,51 @@ possible for the site to automatically trigger it. Second, the capability to foc
 window on its own domain and fire an event containing the clicked URL. This is intended to allow the
 site to navigate an existing window to a new page, overriding the default HTML navigation flow.
 
+## Migrate to Launch Handler API {: #migration }
+
+The Declarative Link Capturing API [origin trial](https://developer.chrome.com/origintrials/#/view_trial/4285175045443026945)
+is set to [expire on March 30, 2022](https://groups.google.com/a/chromium.org/g/blink-dev/c/2c4bul4V3GQ/m/Anluh1txBQAJ)
```
Comment (tomayac) [inline src/site/content/en/blog/declarative-link-capturing/index.md:160]:
```suggestion
is set to [expire on March&nbsp;30, 2022](https://groups.google.com/a/chromium.org/g/blink-dev/c/2c4bul4V3GQ/m/Anluh1txBQAJ)
```

**Review by tomayac — APPROVED** [review]
LGTM, approval

Thanks!

**Comment by tomayac** [comment]
@jpmedley: This PR is factually correct and has  been reviewed by me. Feel free to merge (or make any stylistic changes you may like). Thanks!

File: src/site/content/en/blog/declarative-link-capturing/index.md [hunk 13]
```diff
@@ -152,6 +154,58 @@ possible for the site to automatically trigger it. Second, the capability to foc
...
+    and place the origin trial key into your web app.
+1.  Add a `"launch_handler"` entry to your site's manifest.
+
+    - If you are using `"capture_links": "new-client"`, then
+      add:`"launch_handler": { "route_to": "new-client" }`
+    - If you are using `"capture_links": "existing-client-navigate"`, then add:
+      `"launch_handler": { "route_to": "existing-client" }`
+    - If you wanted to use `"capture_links": "existing-client-event"` (which was never implemented
+      in the Declarative Link Capturing origin trial), then add:
+      `json { "launch_handler": { "route_to": "existing-client", "navigate_existing_client": "never" } } `
```
Comment (jpmedley) [inline src/site/content/en/blog/declarative-link-capturing/index.md:200]: Why is this one phrased differently?

>Which was never implemented...

Why is this here? That we didn't implement it in the link capture trial isn't really relevant to the launch handler trial. Drawing attention to the fact that we didn't implement it could give the impression that we still regard it as a second class citizen.

File: src/site/content/en/blog/declarative-link-capturing/index.md [hunk 14]
```diff
@@ -152,6 +154,58 @@ possible for the site to automatically trigger it. Second, the capability to foc
...
+The control of an incoming navigation is migrated to Launch Handler API, which allows web apps to
+decide how a web app launches at various situations, such as link capturing, share target or file
+handling, etc. To migrate from the Declarative Link Capturing API to the Launch Handler API:
+
+1.  Register your site for the
+    [Launch Handler origin trial](https://developer.chrome.com/origintrials/#/view_trial/2978005253598740481)
+    and place the origin trial key into your web app.
+1.  Add a `"launch_handler"` entry to your site's manifest.
+
+    - If you are using `"capture_links": "new-client"`, then
```
Comment (jpmedley) [inline src/site/content/en/blog/declarative-link-capturing/index.md:194]:
```suggestion
    - To use `"capture_links": "new-client"`,
```

Comment (jpmedley) [reply src/site/content/en/blog/declarative-link-capturing/index.md:194]: Makes the list quicker and easier to scan by reducing duplicate information.

File: src/site/content/en/blog/declarative-link-capturing/index.md [hunk 15]
```diff
@@ -152,6 +154,58 @@ possible for the site to automatically trigger it. Second, the capability to foc
...
+      add:`"launch_handler": { "route_to": "new-client" }`
+    - If you are using `"capture_links": "existing-client-navigate"`, then add:
```
Comment (jpmedley) [inline src/site/content/en/blog/declarative-link-capturing/index.md:196]:
```suggestion
    - To use `"capture_links": "existing-client-navigate"`, add:
```

File: src/site/content/en/blog/declarative-link-capturing/index.md (hunk 13 above)
Comment (jpmedley) [reply src/site/content/en/blog/declarative-link-capturing/index.md:200]: This one will need to change to match the other two.

File: src/site/content/en/blog/declarative-link-capturing/index.md [hunk 16]
```diff
@@ -152,6 +154,58 @@ possible for the site to automatically trigger it. Second, the capability to foc
...
+      `"launch_handler": { "route_to": "existing-client" }`
+    - If you wanted to use `"capture_links": "existing-client-event"` (which was never implemented
+      in the Declarative Link Capturing origin trial), then add:
+      `json { "launch_handler": { "route_to": "existing-client", "navigate_existing_client": "never" } } `
+            **Note**: With this option, pages in your app scope will no longer navigate automatically when a link navigation is captured. You must handle the `LaunchParams` in JavaScript via the `window.launchQueue.setConsumer()` API to enable navigation.
```
Comment (jpmedley) [inline src/site/content/en/blog/declarative-link-capturing/index.md:201]:
```suggestion
            **Note**: With this option, pages in your app scope will no longer navigate automatically when a link navigation is captured. You must handle the `LaunchParams` in JavaScript by calling `window.launchQueue.setConsumer()` to enable navigation.
```

File: src/site/content/en/blog/declarative-link-capturing/index.md [hunk 17]
```diff
@@ -152,6 +154,58 @@ possible for the site to automatically trigger it. Second, the capability to foc
...
+
+1.  Keep the `capture_links` field and Declarative Link Capturing origin trial registration until
+    March&nbsp;30, 2022. This will ensure users on Chromium&nbsp;97 and below can still launch the
+    web app at a captured link.
```
[the remark this answers is not in the document]
Comment (jpmedley) [reply src/site/content/en/blog/declarative-link-capturing/index.md:205]: Phrased as a step, this doesn't say what's important about March 30. It implies that this cut off date is their choice and not ours.

**Review by jpmedley — CHANGES_REQUESTED** [review]

File: src/site/content/en/blog/declarative-link-capturing/index.md [hunk 18]
```diff
@@ -9,7 +9,7 @@ subhead: |
 authors:
   - thomassteiner
 date: 2021-05-19
-updated: 2021-12-14
+updated: 2021-12-21
```
Comment (tomayac) [inline src/site/content/en/blog/declarative-link-capturing/index.md:12]:
```suggestion
updated: 2021-12-22
```

File: src/site/content/en/blog/declarative-link-capturing/index.md (hunk 13 above)
Comment (tomayac) [reply src/site/content/en/blog/declarative-link-capturing/index.md:200]: I think the rhythm was broken since this was never usable.

Comment (jpmedley) [reply src/site/content/en/blog/declarative-link-capturing/index.md:199]: I don't follow.

Comment (tomayac) [reply src/site/content/en/blog/declarative-link-capturing/index.md]: Aligned the rhythm. This should work now.

**Comment by tomayac** [comment]
@jpmedley PTAL now. Thanks!

File: src/site/content/en/blog/declarative-link-capturing/index.md (hunk 13 above)
Comment (jpmedley) [reply src/site/content/en/blog/declarative-link-capturing/index.md]: The rhythm is right. (The term of art among writers is parallelism.) 

For this to stay, it should be in positive form. 'New in this version' as opposed to 'not previously implemented'. We care that it's here now, not that it wasn't implemented before.

Comment (tomayac) [reply src/site/content/en/blog/declarative-link-capturing/index.md]: This is a statement about the past, though. The explainer said it was there and the article documented it, but the implementation never supported it. The goal of this paragraph is to steer people away from the old stuff (declarative link capturing) and toward the new stuff (launch handler). So we want the old stuff to look worse than the new stuff.

Comment (jpmedley) [reply src/site/content/en/blog/declarative-link-capturing/index.md]: I know it's about the past. I'm saying it shouldn't be.

Comment (tomayac) [reply src/site/content/en/blog/declarative-link-capturing/index.md]: Then please change it so that the sense is preserved.

**Review by jpmedley — APPROVED** [review]
